### PR TITLE
Add Javadocs for Monitor Query

### PR DESCRIPTION
**Pull request by g2vinay**
Revamps Javadocs for Azure Monitor Query.

### CODE REVIEW

**Review by jairmyree — APPROVED** [review]
Thank you @g2vinay

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java [hunk 1]
```diff
@@ -37,8 +37,16 @@
 import static com.azure.monitor.query.implementation.logs.models.LogsQueryHelper.mapLogsQueryError;
 
 /**
- * The asynchronous client for querying Azure Monitor logs.
- * <p><strong>Instantiating an asynchronous Logs query Client</strong></p>
+ * <p>Provides an asynchronous service client for querying logs in the Azure Monitor Service.</p>
+ *
+ * <p>The LogsQueryClient is an asynchronous client that provides methods to execute Kusto queries against
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java:42]: `LogsQueryAsyncClient`

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryClientBuilder.java [hunk 2]
```diff
@@ -21,9 +21,23 @@
 import com.azure.monitor.query.implementation.logs.AzureLogAnalyticsImplBuilder;
 
 /**
- * Fluent builder for creating instances of {@link LogsQueryClient} and {@link LogsQueryAsyncClient}.
+ * <p>Fluent builder for creating instances of {@link LogsQueryClient} and {@link LogsQueryAsyncClient}.</p>
  *
- * <p><strong>Instantiating an asynchronous Logs query Client</strong></p>
+ * <p>The LogsQueryClientBuilder is responsible for authenticating a building instances of {@link LogsQueryClient} and
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryClientBuilder.java:26]: `for authenticating` should be removed

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryClientBuilder.java [hunk 3]
```diff
@@ -33,7 +47,7 @@
  * </pre>
  * <!-- end com.azure.monitor.query.LogsQueryAsyncClient.instantiation -->
  *
- * <p><strong>Instantiating a synchronous Logs query Client</strong></p>
+ * <p>The following sample shows instantiating a synchronous Logs query Client using Token Credential</p>
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryClientBuilder.java:50]: `Client` -> `client` and `using Token Credential` should be updated to `using {@link TokenCredential}`

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java [hunk 4]
```diff
@@ -33,8 +33,19 @@
...
 
 /**
- * The asynchronous client for querying Azure Monitor metrics.
- * <p><strong>Instantiating an asynchronous Metrics query Client</strong></p>
+ * <p>The asynchronous client for querying Azure Monitor metrics.</p>
+ *
+ * <p>Azure Monitor Metrics is a feature of Azure Monitor that collects numeric data from monitored resources into a
+ * time-series database. Metrics are numerical values that are collected at regular intervals and describe some aspect
+ * of a system at a particular time. The MetricsQueryClient provides synchronous implementations of methods that query
+ * metrics from your Azure services.</p>
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java:41]: This should be about the async client.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java [hunk 5]
```diff
@@ -33,8 +33,19 @@
...
+ *
+ * <h2>Getting Started</h2>
+ *
+ * <p>
+ *     Authenticating and building MetricsQueryAsyncClient instances are done through {@link MetricsQueryClientBuilder}.
+ *     The following sample shows how to build a new MetricsQueryClient instance.
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java:47]: `MetricsQueryClient` -> `MetricsQueryAsyncClient`

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MetricsQueryResourcesOptions.java [hunk 6]
```diff
@@ -20,8 +20,12 @@ public final class MetricsQueryResourcesOptions {
     private String orderBy;
     private String filter;
 
-    private String rollupBy;
+    /**
+     * Creates an instance of MetricsQueryResourcesOptions.
+     */
+    public MetricsQueryResourcesOptions() { }
 
+    private String rollupBy;
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/models/MetricsQueryResourcesOptions.java:28]: The constructor should be below this property.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java [hunk 7]
```diff
@@ -2,6 +2,121 @@
 // Licensed under the MIT License.
 
 /**
- * Package containing clients for querying logs and metrics from Azure Monitor.
+ * <p>Azure Monitor Query service is a powerful tool that allows you to query and analyze log data from various sources
+ * in Azure. It is built on top of the Kusto Query Language (KQL), which is a powerful query language that allows you
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java:6]: Only logs query uses Kusto Query Language. Metrics queries don't. So, we should update this doc to differentiate.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java [hunk 8]
```diff
@@ -2,6 +2,121 @@
...
+ * to perform complex queries on large datasets. With Azure Monitor Query, you can easily search and analyze
+ * log data from various sources, including virtual machines, containers, and applications.</p>
+ *
+ * <p>Azure Monitor Query java client library is a library that allows you to execute read-only queries against
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java:10]: `Azure Monitor Query java client library allows you to execute`

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java [hunk 9]
```diff
@@ -2,6 +2,121 @@
...
+ *     <li><a href="https://learn.microsoft.com/azure/azure-monitor/essentials/data-platform-metrics">Metrics</a> - Collects
+ *     numeric data from monitored resources into a time series database. Metrics are numerical values that are
+ *     collected at regular intervals and describe some aspect of a system at a particular time. Metrics are lightweight
+ *     and capable of supporting near real-time scenarios, making them particularly useful for alerting and fast
+ *     detection of issues.</li>
+ * </ul>
+ *
+ * <h2>Getting Started</h2>
+ *
+ * <p>In order to interact with the Monitor service you'll need to create an instance of the
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java:30]: `interact with the Azure Monitor service`

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java [hunk 10]
```diff
@@ -2,6 +2,121 @@
...
+ * LogsQueryResult queryResult = logsQueryClient.queryWorkspace&#40;&quot;&#123;workspace-id&#125;&quot;, &quot;&#123;kusto-query&#125;&quot;,
+ *         QueryTimeInterval.LAST_DAY&#41;;
+ * for &#40;LogsTableRow row : queryResult.getTable&#40;&#41;.getRows&#40;&#41;&#41; &#123;
+ *     System.out.println&#40;row.getRow&#40;&#41;
+ *             .stream&#40;&#41;
+ *             .map&#40;LogsTableCell::getValueAsString&#41;
+ *             .collect&#40;Collectors.joining&#40;&quot;,&quot;&#41;&#41;&#41;;
+ * &#125;
+ * </pre>
+ * <!-- end com.azure.monitor.query.LogsQueryClient.query#String-String-QueryTimeInterval -->
```
Comment (srnagar) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/package-info.java:106]: We should have a sample for querying metrics too if we are going to add a sample for querying logs here.

**Pull request merged**